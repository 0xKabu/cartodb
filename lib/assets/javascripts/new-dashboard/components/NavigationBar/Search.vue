<template>

</template>

<script>
export default {
  name: 'NavigationBar'
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
@import '../styles/variables';

.navbar {
    width: 100%;
    padding: 0px 64px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    background-color: $primaryColor;
}

.navbar-elementsContainer {
    display: flex;
    align-items: center;
    margin: 0px;
    padding: 0;
}

.navbar-elementItem {
    padding: 20px 0px 16px 0px;
    margin-right: 34px;
    border-bottom: 4px solid transparent;
    display: flex;
    align-items: center;
    &.is-active {
        border-color: $white;
    }
}

.navbar-icon{
    margin-right: 8px;
}

.navbar-imagotype{
    position: absolute;
    height: 24px;
    width: 24px;
    margin-left: 50%;
    transform: translateX(-50%);
}

.navbar-search {
    margin: 0;
    input {
        width: 134px;
        height: 36px;
        background-color: #fff;
        border-radius: 18px;
        border: 0px;
        padding: 0 4px 0 38px;
        background-image: url("/img/icons/navbar/loupe.svg");
        background-repeat: no-repeat;
        background-position: 16px center;
        transition: width .3s cubic-bezier(.4,.01,.165,.99);
        &::placeholder {
            color: $textColor-light;
        }
        &:focus {
            width: 280px;
            outline: none;
        }
    }
}

.navbar-avatar {
    height: 36px;
    width: 36px;
    object-fit: cover;
    overflow: hidden;
    background-color: $textColor-light;
    border-radius: 50%;
    display: flex;
    margin-left: 30px;
    &:hover{
        cursor: pointer;
    }
}
</style>
