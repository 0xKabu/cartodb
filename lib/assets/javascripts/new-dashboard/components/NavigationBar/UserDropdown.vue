<template>
  <div class="navbar-dropdown" ref="injectionHTMLTemplate" v-once></div>
</template>

<script>
import SettingsDropdown from 'dashboard/components/dashboard-header/settings-dropdown-view';

export default {
  name: 'UserDropdown',
  props: {
    configModel: Object,
    open: { type: Boolean, default: false },
    userModel: Object
  },
  watch: {
    open (newVisibility) {
      this.toggle(newVisibility);
    }
  },
  mounted () {
    this.$dropdownView = this.renderView();
  },
  methods: {
    renderView () {
      const settingsDropdown = new SettingsDropdown({
        model: this.$cartoModels.user,
        configModel: this.$cartoModels.config,
        className: 'Dropdown SettingsDropdown vertical_bottom horizontal_right tick_right'
      });

      settingsDropdown.on('onDropdownHidden', () => {
        this.$emit('dropdownHidden');
      });

      settingsDropdown.render();

      this.$refs.injectionHTMLTemplate.appendChild(settingsDropdown.el);

      return settingsDropdown;
    },

    show () {
      this.$dropdownView.show();
    },

    hide () {
      this.$dropdownView.hide();
    },

    toggle (setVisible) {
      if (setVisible) {
        this.show();
      } else if (!setVisible) {
        this.hide();
      }
    }
  }
};
</script>

<style lang="scss">
.SettingsDropdown {
  top: 48px;
  right: 1px;
}
</style>
