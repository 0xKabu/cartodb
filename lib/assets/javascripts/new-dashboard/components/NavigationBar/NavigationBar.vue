<template>
<nav class="navbar" :class="{ 'is-search-open': isSearchOpen }">
  <ul class="navbar-elementsContainer">
      <router-link :to="{ name: 'maps' }" class="navbar-elementItem" active-class="is-active">
        <span class="navbar-icon">
          <img svg-inline class="navbar-iconFill" src="../../assets/icons/navbar/maps.svg" />
        </span>
        <span class="title is-caption is-txtWhite u-hideTablet">Maps</span>
      </router-link>
      <router-link :to="{ name: 'solutions' }" class="navbar-elementItem" active-class="is-active">
        <span class="navbar-icon">
          <img svg-inline class="navbar-iconFill" src="../../assets/icons/navbar/solutions.svg" />
        </span>
        <span class="title is-caption is-txtWhite u-hideTablet">Solutions</span>
      </router-link>
      <router-link :to="{ name: 'datasets' }" class="navbar-elementItem" active-class="is-active">
        <span class="navbar-icon">
          <img svg-inline class="navbar-iconFill" src="../../assets/icons/navbar/data.svg" />
        </span>
        <span class="title is-caption is-txtWhite u-hideTablet">Data</span>
      </router-link>
      <span class="navbar-elementItem navbar-icon navbar-searchIcon u-showMobile" @click="toggleSearch">
        <img svg-inline src="../../assets/icons/navbar/loupe_white.svg" />
      </span>
  </ul>
  <div class="navbar-imagotype u-hideMobile">
      <img src="../../assets/icons/navbar/imagotype.svg" />
  </div>
  <div class="navbar-elementsContainer navbar-searchContainer">
      <form action="#" method="get" class="navbar-search" autocomplete="off">
          <input type="text" name="query" class="title is-small is-regular navbar-searchInput" placeholder="Search">
      </form>
      <div class="navbar-user">
        <div class="navbar-avatar" :class="{'has-notification': notificationsCount}" :style="{ backgroundImage: `url('${user.avatar_url}')` }" @click.stop.prevent="toggleDropdown"></div>
        <UserDropdown :userModel="user" :notificationsCount="notificationsCount" :open="isDropdownOpen" :baseUrl="baseUrl" v-click-outside="closeDropdown" @linkClick="closeDropdown" />
      </div>
      <span class="navbar-searchClose" @click="toggleSearch">
        <img svg-inline src="../../assets/icons/navbar/close.svg" />
      </span>
  </div>
</nav>
</template>

<script>
import UserDropdown from './UserDropdown';

export default {
  name: 'NavigationBar',
  components: {
    UserDropdown
  },
  props: {
    user: Object,
    baseUrl: String,
    notificationsCount: Number
  },
  data () {
    return {
      isDropdownOpen: false,
      isSearchOpen: false
    };
  },
  methods: {
    toggleDropdown () {
      this.isDropdownOpen = !this.isDropdownOpen;
    },

    closeDropdown () {
      console.log('closeDropdown');
      this.isDropdownOpen = false;
    },

    toggleSearch () {
      this.isSearchOpen = !this.isSearchOpen;
    }
  }
};
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="scss">
@import 'stylesheets/new-dashboard/variables';

.navbar {
  display: flex;
  position: fixed;
  z-index: 3;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  padding: 0 64px;
  background-color: $primary-color;

  @media (max-width: $layout-tablet) {
    padding: 0 24px 0 40px;
  }

  @media (max-width: $layout-mobile) {
    padding: 0 16px 0 20px;
  }
}

.navbar-elementsContainer {
  display: flex;
  position: relative;

  a:last-of-type {
    margin-right: 0;
  }

  @media (max-width: $layout-mobile) {
    a:last-of-type {
      margin-right: 36px;
    }
  }
}

.navbar-elementItem {
  display: flex;
  align-items: center;
  margin-right: 54px;
  padding: 20px 2px 16px;
  border-bottom: 4px solid transparent;

  @media (max-width: $layout-tablet) {
    margin-right: 48px;
  }

  @media (max-width: $layout-mobile) {
    margin-right: 36px;
  }

  &.is-active {
    border-color: $white;

    .navbar-iconFill {
      fill: #FFF;
    }

    .navbar-iconNegativeFill {
      fill: transparent;
    }
  }
}

.navbar-searchIcon {
  padding-top: 21px;
}

.navbar-searchClose {
  display: none;
}

.navbar-icon {
  margin-right: 12px;

  @media (max-width: $layout-tablet) {
    margin-right: 0;
  }
}

.navbar-imagotype {
  width: 24px;
  height: 24px;
}

.navbar-search {
  @media (max-width: $layout-mobile) {
    visibility: hidden;
    opacity: 0;
  }

  &::after {
    content: '';
    position: absolute;
    top: 50%;
    left: 16px;
    width: 18px;
    height: 18px;
    transform: translate3d(0, -50%, 0);
    background-image: url("../../assets/icons/navbar/loupe.svg");
    background-repeat: no-repeat;
    background-position: center;
  }

  &:hover {
    &::after {
      background-image: url("../../assets/icons/navbar/loupe_primary.svg");
    }

    .navbar-searchInput {
      &::placeholder {
        color: $primary-color;
      }

      &:focus {
        &::placeholder {
          color: $text-secondary-color;
        }
      }
    }
  }

  &:focus-within {
    &::after {
      background-image: url("../../assets/icons/navbar/loupe.svg");
    }
  }
}

.navbar-searchInput {
  width: 120px;
  height: 36px;
  padding-left: 42px;
  transition: width 0.3s cubic-bezier(0.4, 0.01, 0.165, 0.99);
  border: 0;
  border-radius: 18px;
  background-color: #FFF;

  &::placeholder {
    color: $text-secondary-color;
  }

  &:focus {
    width: 240px;
    outline: none;
  }
}

.navbar-user {
  position: relative;
}

.navbar-avatar {
  display: flex;
  width: 36px;
  height: 36px;
  margin-left: 24px;
  overflow: hidden;
  border-radius: 50%;
  background-color: $text-color-light;
  background-size: cover;

  &.has-notification {
    &::after {
      content: "";
      position: absolute;
      top: -4px;
      right: -4px;
      width: 12px;
      height: 12px;
      border: 2px solid $primary-color;
      border-radius: 50%;
      background-color: $notification;
    }
  }

  &:hover {
    cursor: pointer;
  }
}

.navbar.is-search-open {
  padding: 0;

  .navbar-elementsContainer {
    align-items: center;
  }

  .navbar-elementsContainer,
  .navbar-imagotype,
  .navbar-user {
    visibility: hidden;
    opacity: 0;
  }

  .navbar-searchContainer {
    visibility: visible;
    position: absolute;
    width: 100%;
    opacity: 1;
  }

  .navbar-search {
    visibility: visible;
    position: absolute;
    left: 0;
    width: calc(100% - 66px);
    margin-left: 16px;
    opacity: 1;

    &::after {
      display: none;
    }
  }

  .navbar-searchInput {
    width: 100%;
    padding: 16px;
    border-radius: 3px;
    background-image: none;
  }

  .navbar-searchClose {
    display: block;
    position: absolute;
    right: 16px;
  }
}
</style>
