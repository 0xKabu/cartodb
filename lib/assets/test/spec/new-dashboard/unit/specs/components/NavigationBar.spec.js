import { mount } from '@vue/test-utils';
import NavigationBar from 'new-dashboard/components/NavigationBar/NavigationBar';

describe('NavigationBar.vue', () => {
  let user;
  beforeEach(() => {
    user = {
      avatar_url: '//cartodb-libs.global.ssl.fastly.net/cartodbui/assets/unversioned/images/avatars/avatar_ghost_yellow.png',
      username: 'Test user',
      email: 'user@test.com'
    };
  });

  it('should render regular navbar with user dropdown closed', () => {
    const navigationBar = mount(NavigationBar, {
      propsData: {
        user
      }
    });
    expect(navigationBar).toMatchSnapshot();
  });

  it('should render regular navbar with user dropdown open', () => {
    const navigationBar = mount(NavigationBar, {
      propsData: {
        user: {}
      }
    });
    navigationBar.vm.toggleDropdown();
    expect(navigationBar).toMatchSnapshot();
  });

  it('should render regular navbar with search input open', () => {
    const navigationBar = mount(NavigationBar, {
      propsData: {
        user: {}
      }
    });
    navigationBar.vm.toggleSearch();
    expect(navigationBar).toMatchSnapshot();
  });

  it('should render regular navbar with user dropdown close after having opened it', () => {
    const navigationBar = mount(NavigationBar, {
      propsData: {
        user: {}
      }
    });
    navigationBar.vm.toggleDropdown();
    navigationBar.vm.closeDropdown();
    expect(navigationBar).toMatchSnapshot();
  });
});
