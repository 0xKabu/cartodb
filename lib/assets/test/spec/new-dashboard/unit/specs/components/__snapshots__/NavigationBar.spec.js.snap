// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar.vue should render regular navbar with search input open 1`] = `
<nav class="navbar is-search-open">
  <ul class="navbar-elementsContainer">
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/maps.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Maps</span></router-link>
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/solutions.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Solutions</span></router-link>
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/data.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Data</span></router-link> <span class="navbar-elementItem navbar-icon navbar-searchIcon u-showMobile"><img svg-inline="" src="../../assets/icons/navbar/loupe_white.svg"></span>
  </ul>
  <div class="navbar-imagotype u-hideMobile"><img src="../../assets/icons/navbar/imagotype.svg"></div>
  <div class="navbar-elementsContainer navbar-searchContainer">
    <form action="#" method="get" autocomplete="off" class="navbar-search"><input type="text" name="query" placeholder="Search" class="title is-small is-regular navbar-searchInput"></form>
    <div class="navbar-user">
      <div class="navbar-avatar" style="background-image: url(undefined);"></div>
      <div class="navbar-dropdown">
        <ul class="navbar-dropdown-container">
          <li class="navbar-dropdown-profile">
            <div class="avatar-container"><img></div>
            <div class="navbar-dropdown-userInfo">
              <p class="text is-semibold is-caption"></p>
              <p class="text is-small"></p>
            </div>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/settings.svg"></div> <a href="/profile" class="text is-semibold is-caption is-txtGrey">Settings</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/api-keys.svg"></div> <a href="/your_apps" class="text is-semibold is-caption is-txtGrey">API Keys &amp; OAuth</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/profile.svg"></div> <a href="/me" class="text is-semibold is-caption is-txtGrey">Public Profile</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/notifications.svg"></div>
            <router-link to="[object Object]" class="text is-semibold is-caption is-txtGrey">Notifications</router-link>
            <!---->
          </li>
        </ul>
        <div class="navbar-dropdown-bottom"><a href="/logout" class="text is-semibold is-caption is-txtGrey">Logout</a></div>
      </div>
    </div> <span class="navbar-searchClose"><img svg-inline="" src="../../assets/icons/navbar/close.svg"></span>
  </div>
</nav>
`;

exports[`NavigationBar.vue should render regular navbar with user dropdown close after having opened it 1`] = `
<nav class="navbar">
  <ul class="navbar-elementsContainer">
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/maps.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Maps</span></router-link>
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/solutions.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Solutions</span></router-link>
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/data.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Data</span></router-link> <span class="navbar-elementItem navbar-icon navbar-searchIcon u-showMobile"><img svg-inline="" src="../../assets/icons/navbar/loupe_white.svg"></span>
  </ul>
  <div class="navbar-imagotype u-hideMobile"><img src="../../assets/icons/navbar/imagotype.svg"></div>
  <div class="navbar-elementsContainer navbar-searchContainer">
    <form action="#" method="get" autocomplete="off" class="navbar-search"><input type="text" name="query" placeholder="Search" class="title is-small is-regular navbar-searchInput"></form>
    <div class="navbar-user">
      <div class="navbar-avatar" style="background-image: url(undefined);"></div>
      <div class="navbar-dropdown">
        <ul class="navbar-dropdown-container">
          <li class="navbar-dropdown-profile">
            <div class="avatar-container"><img></div>
            <div class="navbar-dropdown-userInfo">
              <p class="text is-semibold is-caption"></p>
              <p class="text is-small"></p>
            </div>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/settings.svg"></div> <a href="/profile" class="text is-semibold is-caption is-txtGrey">Settings</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/api-keys.svg"></div> <a href="/your_apps" class="text is-semibold is-caption is-txtGrey">API Keys &amp; OAuth</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/profile.svg"></div> <a href="/me" class="text is-semibold is-caption is-txtGrey">Public Profile</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/notifications.svg"></div>
            <router-link to="[object Object]" class="text is-semibold is-caption is-txtGrey">Notifications</router-link>
            <!---->
          </li>
        </ul>
        <div class="navbar-dropdown-bottom"><a href="/logout" class="text is-semibold is-caption is-txtGrey">Logout</a></div>
      </div>
    </div> <span class="navbar-searchClose"><img svg-inline="" src="../../assets/icons/navbar/close.svg"></span>
  </div>
</nav>
`;

exports[`NavigationBar.vue should render regular navbar with user dropdown closed 1`] = `
<nav class="navbar">
  <ul class="navbar-elementsContainer">
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/maps.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Maps</span></router-link>
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/solutions.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Solutions</span></router-link>
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/data.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Data</span></router-link> <span class="navbar-elementItem navbar-icon navbar-searchIcon u-showMobile"><img svg-inline="" src="../../assets/icons/navbar/loupe_white.svg"></span>
  </ul>
  <div class="navbar-imagotype u-hideMobile"><img src="../../assets/icons/navbar/imagotype.svg"></div>
  <div class="navbar-elementsContainer navbar-searchContainer">
    <form action="#" method="get" autocomplete="off" class="navbar-search"><input type="text" name="query" placeholder="Search" class="title is-small is-regular navbar-searchInput"></form>
    <div class="navbar-user">
      <div class="navbar-avatar" style="background-image: url(//cartodb-libs.global.ssl.fastly.net/cartodbui/assets/unversioned/images/avatars/avatar_ghost_yellow.png);"></div>
      <div class="navbar-dropdown">
        <ul class="navbar-dropdown-container">
          <li class="navbar-dropdown-profile">
            <div class="avatar-container"><img src="//cartodb-libs.global.ssl.fastly.net/cartodbui/assets/unversioned/images/avatars/avatar_ghost_yellow.png"></div>
            <div class="navbar-dropdown-userInfo">
              <p class="text is-semibold is-caption">Test user</p>
              <p class="text is-small">user@test.com</p>
            </div>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/settings.svg"></div> <a href="/profile" class="text is-semibold is-caption is-txtGrey">Settings</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/api-keys.svg"></div> <a href="/your_apps" class="text is-semibold is-caption is-txtGrey">API Keys &amp; OAuth</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/profile.svg"></div> <a href="/me" class="text is-semibold is-caption is-txtGrey">Public Profile</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/notifications.svg"></div>
            <router-link to="[object Object]" class="text is-semibold is-caption is-txtGrey">Notifications</router-link>
            <!---->
          </li>
        </ul>
        <div class="navbar-dropdown-bottom"><a href="/logout" class="text is-semibold is-caption is-txtGrey">Logout</a></div>
      </div>
    </div> <span class="navbar-searchClose"><img svg-inline="" src="../../assets/icons/navbar/close.svg"></span>
  </div>
</nav>
`;

exports[`NavigationBar.vue should render regular navbar with user dropdown open 1`] = `
<nav class="navbar">
  <ul class="navbar-elementsContainer">
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/maps.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Maps</span></router-link>
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/solutions.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Solutions</span></router-link>
    <router-link to="[object Object]" active-class="is-active" class="navbar-elementItem"><span class="navbar-icon"><img svg-inline="" src="../../assets/icons/navbar/data.svg" class="navbar-iconFill"></span> <span class="title is-caption is-txtWhite u-hideTablet">Data</span></router-link> <span class="navbar-elementItem navbar-icon navbar-searchIcon u-showMobile"><img svg-inline="" src="../../assets/icons/navbar/loupe_white.svg"></span>
  </ul>
  <div class="navbar-imagotype u-hideMobile"><img src="../../assets/icons/navbar/imagotype.svg"></div>
  <div class="navbar-elementsContainer navbar-searchContainer">
    <form action="#" method="get" autocomplete="off" class="navbar-search"><input type="text" name="query" placeholder="Search" class="title is-small is-regular navbar-searchInput"></form>
    <div class="navbar-user">
      <div class="navbar-avatar" style="background-image: url(undefined);"></div>
      <div class="navbar-dropdown is-open">
        <ul class="navbar-dropdown-container">
          <li class="navbar-dropdown-profile">
            <div class="avatar-container"><img></div>
            <div class="navbar-dropdown-userInfo">
              <p class="text is-semibold is-caption"></p>
              <p class="text is-small"></p>
            </div>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/settings.svg"></div> <a href="/profile" class="text is-semibold is-caption is-txtGrey">Settings</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/api-keys.svg"></div> <a href="/your_apps" class="text is-semibold is-caption is-txtGrey">API Keys &amp; OAuth</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/profile.svg"></div> <a href="/me" class="text is-semibold is-caption is-txtGrey">Public Profile</a>
          </li>
          <li class="navbar-dropdown-iconLink">
            <div class="icon-container"><img src="../../assets/icons/navbar/dropdown/notifications.svg"></div>
            <router-link to="[object Object]" class="text is-semibold is-caption is-txtGrey">Notifications</router-link>
            <!---->
          </li>
        </ul>
        <div class="navbar-dropdown-bottom"><a href="/logout" class="text is-semibold is-caption is-txtGrey">Logout</a></div>
      </div>
    </div> <span class="navbar-searchClose"><img svg-inline="" src="../../assets/icons/navbar/close.svg"></span>
  </div>
</nav>
`;
